packages/tizen-remote/test/e2e/tizen-remote.e2e.spec.js

```diff
@@ -0,0 +1,343 @@
+
+const expect = unexpected.clone().use(unexpectedSinon).use(unexpectedEventEmitter);
+
+describe('websocket behavior', function () {
+  /** @type {TestWSServer} */
+  let server;
+
+  /** @type {number} */
+  let port;
+
+  /** @type {TizenRemote} */
+  let remote;
+
+  /** @type {sinon.SinonSandbox} */
+  let sandbox;
+
+  /** @type {import('../../lib').TizenRemoteOptions} */
+  let remoteOpts;
+
+  beforeEach(async function () {
+    sandbox = createSandbox();
+    if (PORT) {
+      port = Number(PORT);
+    } else {
+      port = await getPort();
+      server = new TestWSServer({
+        host: HOST,
+        port,
+        path: constants.API_PATH_V2,
+      });
+    }
+    remoteOpts = {
+      host: HOST,
+      port,
+      name: 'test',
+      token: 'hoboken',
+    };
+  });
+
+  afterEach(async function () {
+    sandbox.restore();
+    if (remote) {
+      try {
+        debug('[CLEANUP] Disconnecting from remote attached to %s:%d', HOST, port);
+        await remote.disconnect();
+      } catch {}
+    }
+    debug('[CLEANUP] Stopping server listening on %s:%d', HOST, port);
+    await server.stop();
+  });
+
+  it('should connect', async function () {
+    remote = new TizenRemote(remoteOpts);
+    await remote.connect();
+    expect(remote.isConnected, 'to be true');
+  });
+
+  describe('connection behavior', function () {
+    describe('token negotiation', function () {
+      describe('when the remote has no token', function () {
+        beforeEach(function () {
+          remoteOpts.token = undefined;
+          remote = new TizenRemote(remoteOpts);
+        });
+
+        it('should request a token from the server', async function () {
+          return await expect(
+            remote.connect(),
+            'to emit from',
+            remote,
+            Event.TOKEN,
+            `token-${remoteOpts.name}`
+          );
+        });
+
+        describe('when the request time exceeds "tokenTimeout"', function () {
+          beforeEach(function () {
+            remoteOpts.token = undefined;
+            remoteOpts.tokenTimeout = 1;
+            remote = new TizenRemote(remoteOpts);
+          });
+          it('should reject', async function () {
+            return await expect(
+              remote.connect(),
+              'to be rejected with error satisfying',
+              /did not receive token in 1ms/i
+            );
+          });
+        });
+      });
+
+      describe('when the remote has a token', function () {
+        beforeEach(function () {
+          remote = new TizenRemote(remoteOpts);
+        });
+        it('should not request a token from the server', async function () {
+          return await expect(remote.connect(), 'not to emit from', remote, Event.TOKEN);
+        });
+      });
+    });
+
+    describe('when connection fails', function () {
+      beforeEach(async function () {
+        // stop the currently-listening server entirely
+        await server.stop();
```

Comment:
allowing this test file to take an optional host name and port makes me think these tests are optionally to be run against a real tizen tv, but the tests themselves are not written that way. so why allow anything but the mock server?

Reply:
I just didn't do that yet. 😄

Reply:
This is done now